packages/salesforcedx-vscode-apex/src/views/testOutline.ts

```diff
@@ -0,0 +1,666 @@
+            this.head.children.push(newApexTestGroup);
+          }
+        }
+      });
+    }
+    return this.head;
+  }
+
+  private addTests(fileContent: string, apexTestGroup: ApexTestGroup): void {
+    // Parse through file and find apex tests
+    const regEx = /(@[iI][sS][tT][eE][sS][tT])+/g;
+    let match = regEx.exec(fileContent);
+    while (match) {
+      const ind = match.index;
+      // Ignore class heading
+      if (ind !== 0) {
+        // A test
+        const restOfFile = fileContent.slice(match.index);
+        const header = restOfFile.slice(0, restOfFile.indexOf('{'));
+        const headerParts = header.split(' ');
+
+        // Name is last part of the header, not always the last item though, because could be an empty string ''
+        let name = '';
+        while (name === '') {
+          const headerPart = headerParts.pop();
+          if (headerPart && headerPart.includes(')')) {
+            name = headerPart;
+          }
+        }
+        name = name.replace('()', ''); // Get rid
+        const apexTest = new ApexTest(name, apexTestGroup.file, ind);
+        apexTest.name = apexTestGroup.label + '.' + name;
+        this.apexTestMap.set(apexTest.name, apexTest);
+        apexTestGroup.children.push(apexTest);
+      }
+      match = regEx.exec(fileContent);
+    }
+  }
+
+  private deleteFolderRecursive(path: string) {
+    let files = [];
+    if (fs.existsSync(path)) {
+      files = fs.readdirSync(path);
+      files.forEach((file, index) => {
+        const curPath = path + '/' + file;
+        if (fs.lstatSync(curPath).isDirectory()) {
+          // recurse
+          this.deleteFolderRecursive(curPath);
+        } else {
+          // delete file
+          fs.unlinkSync(curPath);
+        }
+      });
+      fs.rmdirSync(path);
+    }
+  }
+
+  public async showErrorMessage(test: Test) {
+    let position = test.row;
+    let isRow = this.apexTestInfo !== null; // If I derived info from apex-jorje, then it is the row, otherwise it is a positional index
+    if (test instanceof ApexTest) {
+      const errorMessage = test.errorMessage;
+      if (errorMessage && errorMessage !== '') {
+        const stackTrace = test.stackTrace;
+        position = parseInt(
+          stackTrace.substring(
+            stackTrace.indexOf('line') + 4,
+            stackTrace.indexOf(',')
+          ),
+          10
+        );
+        isRow = true;
+        channelService.appendLine(
+          '-----------------------------------------------------------'
+        );
+        channelService.appendLine(stackTrace);
+        channelService.appendLine(errorMessage);
+        channelService.appendLine(
+          '-----------------------------------------------------------'
+        );
+        channelService.showChannelOutput();
+      }
+    }
+
+    if (test.file) {
+      vscode.window.showTextDocument(test.file).then(() => {
+        this.eventsEmitter.emit('Show Highlight', position, isRow);
```

Comment:
@jkalloor0701 - All messages should be externalized.
@ruthemmanuelle - Comments?

Reply:
@jkalloor0701 - Like other messages, to provide feedback I'd like to know:
1. When users will encounter this message.
2. Where and how the message will be displayed.
And, yes, all messages should be in a messages file. That way we'll be able to translate them if we localize the extension pack, and it's easier for me to keep an eye on any changes to the messages that people make in the future.

Reply:
@jkalloor0701 - Just making sure you saw this comment, since you already responded to my other questions.

Reply:
Oh whoops, forgot to respond to this one. This is actually not a message but the event emitter launches a different function to run. @ruthemmanuelle